src/utilities/include/sparse.h

```diff
@@ -110,59 +106,41 @@ class SparseIndices {
     }
 
     __host__ __device__ bool operator()(
-        thrust::tuple<T, int, int, int> x) const {
+        thrust::tuple<T, int, int64_t> x) const {
       bool result = NotFinite(thrust::get<0>(x));
       return result;
     }
   };
 
   template <typename T>
   size_t KeepFinite(VecDH<T>& v, VecDH<int>& x) {
-    ASSERT(x.size() == p.size(), userErr,
+    ASSERT(x.size() == data.size(), userErr,
            "Different number of values than indicies!");
-    auto zBegin = zip(v.begin(), x.begin(), begin(false), begin(true));
-    auto zEnd = zip(v.end(), x.end(), end(false), end(true));
+    auto zBegin = zip(v.begin(), x.begin(), data.begin());
+    auto zEnd = zip(v.end(), x.end(), data.end());
     size_t size = remove_if<decltype(zBegin)>(autoPolicy(v.size()), zBegin,
                                               zEnd, firstNonFinite<T>()) -
                   zBegin;
     v.resize(size);
-    x.resize(size, -1);
-    p.resize(size, -1);
-    q.resize(size, -1);
+    x.resize(size);
+    Resize(size);
     return size;
   }
 
-  template <typename Iter, typename T>
-  VecDH<T> Gather(const VecDH<T>& val, const Iter pqBegin, const Iter pqEnd,
-                  T missingVal) const {
-    ASSERT(val.size() == p.size(), userErr,
-           "Different number of values than indicies!");
-    size_t size = pqEnd - pqBegin;
-    VecDH<T> result(size);
-    VecDH<char> found(size);
-    VecDH<int> temp(size);
-    auto policy = autoPolicy(size);
-    fill(policy, result.begin(), result.end(), missingVal);
-    binary_search(policy, beginPQ(), endPQ(), pqBegin, pqEnd, found.begin());
-    lower_bound(policy, beginPQ(), endPQ(), pqBegin, pqEnd, temp.begin());
-    gather_if(policy, temp.begin(), temp.end(), found.begin(), val.begin(),
-              result.begin());
-    return result;
-  }
-
 #ifdef MANIFOLD_DEBUG
   void Dump() const {
-    const auto& p = Get(0);
-    const auto& q = Get(1);
     std::cout << "SparseIndices = " << std::endl;
+    const int* p = ptr();
     for (int i = 0; i < size(); ++i) {
-      std::cout << i << ", p = " << p[i] << ", q = " << q[i] << std::endl;
+      std::cout << i << ", p = " << p[2 * i + pOffset]
+                << ", q = " << p[2 * i + 1 - pOffset] << std::endl;
```

Comment:
I've seen these accessors used a lot - should we just define `int Get(int i, bool useQ)` to clean it up?

Reply:
Or even `int Get<useQ>(int i)`

Reply:
I just realized as we disabled cuda, we can just make these functions take a reference to the original SparseIndices and use functions on the SparseIndices instead of doing all those pointer arithmetic. Will fix.